nixos/modules/services/mail/rss2email.nix

```diff
@@ -0,0 +1,221 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.rss2email;
+  conf = pkgs.writeText "rss2email.cfg" ''
+    [DEFAULT]
+    from = ${cfg.from}
+    to = ${cfg.to}
+    ${cfg.configuration}
+
+    ${builtins.concatStringsSep "\n" (
+      builtins.map (name: let feed = cfg.feeds.${name}; in ''
+        [feed.${name}]
+        url = ${feed.url}
+        ${lib.optionalString (feed.to != null) "to = ${feed.to}"}
+      '') (builtins.attrNames cfg.feeds)
+    )}
+  '';
+
+in {
+
+  ###### interface
+
+  options = {
+
+    services.rss2email = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable rss2email.";
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.rss2email;
+        defaultText = "pkgs.rss2email";
+        description = "The rss2email package to use.";
+      };
+
+      from = mkOption {
+        type = types.str;
+        default = "user@rss2email.invalid";
+        description = "Mail address to use to send emails";
+      };
+
+      to = mkOption {
+        type = types.str;
+        description = "Mail address to which to send emails";
+      };
+
+      interval = mkOption {
+        type = types.str;
+        default = "1h";
+        description = "How often to check the feeds, in systemd interval format";
+      };
+
+      configuration = mkOption {
```

Comment:
You could use https://github.com/NixOS/nixpkgs/blob/master/lib/generators.nix and take an attrset instead.